### PR TITLE
support pgx v5

### LINKED ISSUE #88: Are there any plans to support pgx v5?
In a month or so, `pgx v5` will be officially released.

* https://github.com/jackc/pgx/issues/1169
* https://github.com/jackc/pgx/issues/1061#issuecomment-1186250809
  >This is the last major change planned for v5. I plan to release the (hopefully) final alpha release within the next few days (maybe even tonight), then a beta release a couple weeks after that, and the general release a few weeks after that.

Here are the commits I supported v5 for reference.

* https://github.com/jnst/scany/commits/master

`pgx v5` **requires** `go 1.18` or higher. do you plan to support this in `scany`?

### PR DESCRIPTION
**Pull request by vadimi**
since pgx v5 contains breaking changes (supports go 1.18+ only) this package also needs to have a version bump - v2

should fix #88

### CODE REVIEW

File: dbscan/dbscan_test.go [hunk 1]
```diff
@@ -9,12 +9,12 @@ import (
...
 	"github.com/jackc/pgtype"
-	"github.com/jackc/pgx/v4/pgxpool"
-	_ "github.com/jackc/pgx/v4/stdlib"
+	"github.com/jackc/pgx/v5/pgxpool"
+	_ "github.com/jackc/pgx/v5/stdlib"
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
 
-	"github.com/georgysavva/scany/dbscan"
+	"github.com/georgysavva/scany/v2/dbscan"
```
Comment (lzap) [inline dbscan/dbscan_test.go:17]: Aren't you missing this directory from the patch?

Comment (vadimi) [reply dbscan/dbscan_test.go:17]: not really, v2 is the version of the module, so it doesn't need to have a directory

Comment (georgysavva) [reply dbscan/dbscan_test.go:17]: Agree with @vadimi

**Comment by georgysavva** [comment]
@vadimi Thank you for your work. It looks great! 

I think we also need to update the readme file to use the new module name

**Comment by vadimi** [comment]
@georgysavva thanks, that makes sense, updated the readme. I also noticed golangci lint is failing, so I updated it to latest to see if it helps.

File: dbscan/dbscan_test.go (hunk 1 above)
Comment (lzap) [reply dbscan/dbscan_test.go:17]: Thanks, I am still fighting with the new module system :-)

**Comment by lzap** [comment]
Thanks for this, eagerly awaiting this to be released :-)

**Comment by darkliquid** [comment]
Looking forward to seeing this released - been on a big OpenTelemetry push recently, so will be nice to be able to use the new support in pgx5 for that.

In the mean time, I've been using a replace directive in my `go.mod` to allow me to use this PR.

```
replace github.com/georgysavva/scany/v2 => github.com/vadimi/scany/v2 v2.0.0-20220923222004-64484fa02b2c
```

**Comment by lzap** [comment]
Hands up who's reason is OpenTelemetry too! :-) Thanks, nice trick actually. Had no idea it works remotely too.

**Comment by georgysavva** [comment]
@vadimi Thanks for updating the readme!

Yes, I also noticed that golangci-lint is broken. I fixed it in the master, along with other GitHub action bumps. So I just merged the changes into your branch. I also fixed a few leftovers for the old version paths.

Now the PR is ready. Thank you so much for taking on it! I am going to proceed with merging it and releasing the v2 tag.

**Comment by georgysavva** [comment]
Here everyone. I created a pre-release for `scany/v2`: https://github.com/georgysavva/scany/releases/tag/v2.0.0-alpha.2. You can install it the following way: `go get https://github.com/georgysavva/scany/v2@v2.0.0-alpha.2`.

Please test it out a little bit, and if everything is fine, I will make the stable release.

**Pull request merged**